compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1823,13 +1822,15 @@ private void checkRecordVarRefEquivalency(DiagnosticPos pos, BLangRecordVarRef l
         BRecordType rhsRecordType = (BRecordType) rhsType;
 
         // check if all fields in record var ref are found in rhs record type
-        lhsVarRef.recordRefFields.stream()
-                .filter(lhsField -> rhsRecordType.fields.stream()
-                        .noneMatch(rhsField -> lhsField.variableName.value.equals(rhsField.name.toString())))
-                .forEach(lhsField -> dlog.error(pos, DiagnosticCode.INVALID_FIELD_IN_RECORD_BINDING_PATTERN,
-                        lhsField.variableName.value, rhsType));
+        for (BLangRecordVarRefKeyValue lhsField : lhsVarRef.recordRefFields) {
+            if (!rhsRecordType.fields.containsKey(lhsField.variableName.value)) {
+                dlog.error(pos, DiagnosticCode.INVALID_FIELD_IN_RECORD_BINDING_PATTERN,
+                           lhsField.variableName.value, rhsType);
+            }
+        }
 
-        for (BField rhsField : rhsRecordType.fields) {
+        for (BField rhsField : rhsRecordType.fields.values()) {
+            // TOOD 22/04/2020: Refactor this. No need to do this in each iteration.
```

Comment:
Can we fix this in this PR itself?

Reply:
It looks like this is necessary. For example,
```ballerina
type Person record {
    string name;
    int age;
    string country;
};


public function main() {    
    string firstName;
    int personAge;
    int personAge2;
    map<anydata|error> otherDetails = {};

    {name: firstName, age: personAge, age: personAge2, ...otherDetails} = getPerson();
}

function getPerson() returns Person {
    Person person = {
        name: "Peter",
        age: 28,
        country: "Sri Lanka",
        "occupation": "Software Engineer"
    };
    return person;
}
```
I can't recall the reason for this comment, maybe I misread the code :sweat_smile:  Will remove the comment